### PR TITLE
Feature/kaa 521

### PR DESCRIPTION
**Pull request by sashadidukh**
Added possibility to get profile based on endpoint key and on endpoint group id from cassandra and mongodb via REST API. Provided pagination feature when got profile by endpoint group id.

### CODE REVIEW

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/controller/KaaAdminController.java [hunk 1]
```diff
@@ -150,6 +171,90 @@ public void handleKaaAdminServiceException(KaaAdminServiceException ex, HttpServ
...
+     * Gets the endpoint profile body by endpoint group id.
+     */
+    @RequestMapping(value = "endpointProfileBodyByGroupId", method = RequestMethod.GET, produces = APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public EndpointProfilesBodyDto getEndpointProfileBodyByEndpointGroupId(
+            @RequestParam(value = "endpointGroupId") String endpointGroupId,
+            @RequestParam(value = "limit", defaultValue = DEFAULT_LIMIT) String limit,
+            @RequestParam(value = "offset", defaultValue = DEFAULT_OFFSET) String offset,
+            HttpServletRequest request) throws KaaAdminServiceException {
+        EndpointProfilesPageDto endpointProfilesPageDto = kaaAdminService.getEndpointProfileBodyByEndpointGroupId(endpointGroupId, limit, offset);
```
Comment (shpigor) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/controller/KaaAdminController.java]: Why do we use different method and expect EndpointProfilesPageDto object, after that convert to another object?

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/controller/KaaAdminController.java [hunk 2]
```diff
@@ -150,6 +171,90 @@ public void handleKaaAdminServiceException(KaaAdminServiceException ex, HttpServ
...
+        if (endpointProfilesPageDto.hasEndpointBodies()) {
+            PageLinkDto pageLinkDto = createNext(endpointProfilesPageDto.getPageLinkDto(), request);
+            endpointProfilesPageDto.setPageLinkDto(pageLinkDto);
+        }
+        return convertToEndpointProfilesBodyDto(endpointProfilesPageDto);
```
Comment (shpigor) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/controller/KaaAdminController.java]: Use correct object

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java [hunk 3]
```diff
@@ -23,10 +23,13 @@
 import static org.kaaproject.kaa.server.common.thrift.util.ThriftDtoConverter.toGenericDataStruct;
 import static org.kaaproject.kaa.server.common.thrift.util.ThriftDtoConverter.toGenericDto;
 import static org.kaaproject.kaa.server.common.thrift.util.ThriftDtoConverter.toGenericDtoList;
+
 import java.io.IOException;
 import java.security.InvalidParameterException;
 import java.util.*;
```
Comment (shpigor) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java]: optimize imports

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java [hunk 4]
```diff
@@ -161,6 +173,99 @@ public void setPasswordEncoder(PasswordEncoder passwordEncoder) {
...
+    }
+
+    private String getApplicationIdByEndpointGroupId(String endpointGroupId) throws KaaAdminServiceException {
+        EndpointGroupDto endpointGroupDto = getEndpointGroup(endpointGroupId);
+        if (endpointGroupDto == null) {
+            LOG.warn("Endpoint group with id {} is not present in db.", endpointGroupId);
+            throw new IllegalArgumentException("Endpoint group is not present in db.");
+        }
+        String applicationId = endpointGroupDto.getApplicationId();
+        checkApplicationId(applicationId);
```
Comment (shpigor) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java]: Double  check, look at getEndpointGroup() method

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java [hunk 5]
```diff
@@ -161,6 +173,99 @@ public void setPasswordEncoder(PasswordEncoder passwordEncoder) {
...
+        ApplicationDto applicationDto = getApplication(applicationId);
+        checkTenantId(applicationDto.getTenantId());
```
Comment (shpigor) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java]: Double  check, look at getEndpointGroup() method

File: server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java [hunk 6]
```diff
@@ -161,6 +173,99 @@ public void setPasswordEncoder(PasswordEncoder passwordEncoder) {
...
+                        "Requested item was not found!",
+                        ServiceErrorCode.ITEM_NOT_FOUND);
+            }
+            return endpointProfilesPage;
+        } catch (Exception e) {
+            throw Utils.handleException(e);
+        }
+    }
+
+    private String getApplicationIdByEndpointGroupId(String endpointGroupId) throws KaaAdminServiceException {
```
Comment (shpigor) [inline server/admin/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java]: Invalid method, please return EndpointGroupDto

File: server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java [hunk 7]
```diff
@@ -205,10 +208,25 @@ public EndpointGroupDto saveEndpointGroup(EndpointGroupDto endpointGroupDto) {
     }
 
     @Override
+    @Transactional
+    public EndpointProfilesPageDto findEndpointProfileByEndpointGroupId(PageLinkDto pageLink) {
+        validateSqlId(pageLink.getLimit(), "Can't find endpoint group by id. Incorrect limit parameter " + pageLink.getLimit());
+        validateString(pageLink.getOffset(), "Can't find endpoint group by id. Incorrect offset parameter " + pageLink.getOffset());
```
Comment (Shpigor) [inline server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java]: please rename method name validateString()

File: server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java [hunk 8]
```diff
@@ -205,10 +208,25 @@ public EndpointGroupDto saveEndpointGroup(EndpointGroupDto endpointGroupDto) {
...
+        return endpointProfileDao.findByEndpointGroupId(pageLink);
+    }
+
+    @Override
+    @Transactional
+    public EndpointProfilesPageDto findEndpointProfileBodyByEndpointGroupId(PageLinkDto pageLink) {
+        validateSqlId(pageLink.getLimit(), "Can't find endpoint group by id. Incorrect limit parameter " + pageLink.getLimit());
+        validateString(pageLink.getOffset(), "Can't find endpoint group by id. Incorrect offset parameter " + pageLink.getOffset());
```
Comment (shpigor) [inline server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java:223]: please rename method name validateString()

File: server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/EndpointProfileCassandraDao.java [hunk 9]
```diff
@@ -122,19 +197,58 @@ public long getCountByKeyHash(byte[] endpointKeyHash) {
...
+        ByteBuffer[] epKeyHashList =  cassandraEPByAppIdDao.getEPIdsListByAppId(appId);
+        if (epKeyHashList != null) {
+            for (ByteBuffer epKeyHash : epKeyHashList) {
+                removeByKeyHashFromEpByEndpointGroupId(getBytes(epKeyHash));
+            }
+        }
         Statement deleteEpsByAppId = delete().from(EP_BY_APP_ID_COLUMN_FAMILY_NAME).where(eq(EP_BY_APP_ID_APPLICATION_ID_PROPERTY, appId));
         executeBatch(deleteEps, deleteEpsByAppId);
-        LOG.trace("Execute statements {}, {} like batch", deleteEps, deleteEpsByAppId);
+        LOG.trace("Execute statements {}, {}, {} like batch", deleteEps, deleteEpsByAppId);
```
Comment (shpigor) [inline server/common/nosql/cassandra-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/cassandra/dao/EndpointProfileCassandraDao.java]: why do we need third {}?

**Pull request closed without merging**